### ISSUE TITLE
Objects that have a property of type 'Date' have their values changed to be an empty object

### ISSUE THREAD

**Issue by jdcookie** [huynhsamha/js-convert-case#11]
I am using this in conjunction with Prisma

Console logging the result from Prisma results in:
```javascript
{
  Id: '77ec59b4-d7c8-4907-8f61-dc468cdd11ff',
  StartDateTime: 2021-04-16T12:30:00.000Z,
}
```
And if I log out the `typeof` for StartDateTime I get
```
StartDateTime: object
```
So I know that it is a `Date` object.

After running the data through the `camelKeys` function:
```javascript
  const source = camelKeys(raw, { recursive: true, recursiveInArray: true })
```

I end up with this
```javascript
{
  "id": "77ec59b4-d7c8-4907-8f61-dc468cdd11ff",
  "startDateTime": {},
}
```

I am guessing that this is because it is recursively looping through the nested object that I am converting, and the Date object counts as an object, but I think you should consider handling this object in a special way and returning either the Date object or at least it's .toString() result.

**Comment by huynhsamha** [comment]
Hi @jdcookie,

For your case, you can use `{ recursive: false, recursiveInArray: false }` to keep `StartDateTime` as an object Date.

I'll find a way to handle an object when doing recursion.

Thanks for your issue.

**Comment by huynhsamha** [comment]
I have released a [new version 4.1.2](https://github.com/huynhsamha/js-convert-case/releases/tag/v4.1.2) for this issue.

In this release, we can add a new option `keepTypesOnRecursion: any[] = []` to keep value on recursion. With this option, every value on recursion will be checked if it is an instance of any type in `keepTypesOnRecursion`.

Example:

```
{ 
    recursive: true, 
    recursiveInArray: true,
    keepTypesOnRecursion: [Date]
}
```

**Comment by jdcookie** [comment]
Thanks @huynhsamha - it seems to be working nicely.

Just FYI I couldn't use the "recursive: false" options because my real scenario was more complex than the example I provided. It was deeply nested this property.

Thanks for your quick response and fix 👌
